wagtail_wordpress_import/block_builder.py

```diff
@@ -52,13 +59,21 @@ def get_builder_function(self, element):
         returns:
             a function to parse the block from configuration
         """
-        function = [
-            import_string(builder[1]["FUNCTION"])
-            for builder in conf_html_tags_to_blocks()
-            if element.name == builder[0]
-        ]
-        if function:
-            return function[0]
+
+        # registered shortcode custom tags
+        conf_custom_tags = {}
+        for handler in SHORTCODE_HANDLERS:
+            cls = handler()
+            conf_custom_tags[cls.element_name] = handler
```

Comment:
> **Question**: Is this so Bleach doesn't later remove them? Or just so we can fetch the builder method from the Shortcode Handler class?

Yes bleach was escaping the opening < and closing > characters. Then the block builder failed to find the tag with BeautifulSoup

Reply:
> I'm still groking this, but do we run this `get_builder_function` method once per element in the soup. Could we farm out the looping over the SHORTCODE_HANDLERS? It seems repetitive.

I will look again at this.

Reply:
I think this could be done differently. We could change the register function to store a dict and have the ALLOWED_HTML_TAGS and the shortcode construct_block stored there when the shortcode handlers are loaded. This would mean we could store the functions as functions and then just call them I think. That would help with get_builder_function()

But as the ALLOWED_HTML_TAGS config is read every time we'd still need to append the shortcode element_name. It would be the same for the TAGS_TO_PROMOTE config here promote_child_tags()